lib/zendesk-ticket-deduplicator.rb

```diff
@@ -0,0 +1,97 @@
+require_relative 'zendesk-setup.rb'
+
+require 'rest-client'
+require 'json'
+
+# puts "Total Users on the system: #{@client.users.count}"
+minus_1_hr = Time.now - 2 * 3600
+time_minus_1_hr = minus_1_hr.strftime('%Y-%m-%dT%H:%M:%S%z')
+
+puts "time: #{time_minus_1_hr}"
+
+# Change date to 2020-01-01
+search_results = @client.search(:query => "type:user role:end-user -name:Zendesk organization:none created_at>=#{time_minus_1_hr}")
+
+user_count =  search_results.count
+number_of_pages = (user_count.to_f / 100).ceil
+
+puts "Number of User accounts (UIDs): #{user_count}"
+puts "Number of pages of UIDs: #{number_of_pages}"
+puts "Retrieving #{user_count} user accounts, this may take a while"
+
+# Reset name after testing - Set Logfile name
+results_file = "data/2020_user_ids.json"
+
+# Loop through users and write to file in JSON format
+File.open(results_file, "w") do |file|
+  (1..number_of_pages).each do |i|
+    search_results.page(i).each do |user|
+      file.puts(user.to_json)
+    end
+  end
+end
+
+# Setup the log file
+log_file_name = ENV['ZENDESK_LOG_FILE']
+File.open(log_file_name, "w") do |log_file|
+
+  # Open the source file for reading
+  File.readlines(results_file).each do |line|
+    user = JSON.parse(line)
+    user_id = user["id"]
+
+    # Count tickets for this user
+    count = @client.search!(:query => "type:ticket requester:#{user_id} -status:closed -status:solved").count
+    ticket_count = Integer count
+
+    # Message and log the result
+    message = "User ID: #{user_id} has #{ticket_count} ticket/s"
+    log_file.puts message
+    puts message
```

Comment:
You should make this a function; the pattern is repeated several times in this file.

Reply:
Will consider in next update.